### PR TITLE
Added history activity in users portal

### PR DESCRIPTION
**Pull request by areksolek**
HedDOQIQke

### CODE REVIEW

File: layouts/basic/modules/Settings/WebserviceUsers/ListViewSession.tpl [hunk 1]
```diff
@@ -19,7 +19,7 @@
 					{foreach key=LISTVIEW_ENTRY_COLUMN_NAME item=LISTVIEW_ENTRY_COLUMN from=$TABLE_COLUMNS}
 						<td class="noWrap {$WIDTHTYPE}">
 							{if isset($LISTVIEW_ENTRY[$LISTVIEW_ENTRY_COLUMN_NAME]) && !empty($LISTVIEW_ENTRY[$LISTVIEW_ENTRY_COLUMN_NAME])}
-								{\App\Purifier::encodeHtml($LISTVIEW_ENTRY[$LISTVIEW_ENTRY_COLUMN_NAME])}
+								{\App\Purifier::decodeHtml($LISTVIEW_ENTRY[$LISTVIEW_ENTRY_COLUMN_NAME])}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Settings/WebserviceUsers/ListViewSession.tpl]: ?

File: modules/Settings/WebserviceUsers/models/Module.php [hunk 2]
```diff
@@ -106,4 +106,14 @@ public function getSessionViewUrl(): string
 	{
 		return 'index.php?module=' . $this->getName() . '&parent=Settings&view=ListViewSession&typeApi=' . $this->typeApi;
 	}
+
+	/**
+	 * Function to get the url for history activity view of the user.
+	 *
+	 * @return string - url
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/WebserviceUsers/models/Module.php]:
```suggestion
	 * @return string URL
```

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 3]
```diff
@@ -154,11 +154,13 @@ public function getEditFields()
 	/**
 	 * Get user session.
 	 *
+	 * @param string $typeApi
+	 *
 	 * @return array
 	 */
-	public function getUserSession(): array
+	public function getUserSession(string $typeApi): array
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/WebserviceUsers/models/Record.php]:
```suggestion
	public function getUserSession(string $container): array
```

File: modules/Settings/WebserviceUsers/models/Record.php [hunk 4]
```diff
@@ -167,6 +169,24 @@ public function getUserSession(): array
...
 	}
 
+	/**
+	 * Get user session.
+	 *
+	 * @param string $typeApi
+	 *
+	 * @return array
+	 */
+	public function getUserHistoryAccessActivity(string $typeApi): array
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/WebserviceUsers/models/Record.php]:
```suggestion
	public function getUserHistoryAccessActivity(string $container): array
```

File: modules/Settings/WebserviceUsers/views/ListViewSession.php [hunk 5]
```diff
@@ -41,10 +41,11 @@ class Settings_WebserviceUsers_ListViewSession_View extends \App\Controller\Moda
 	public function process(App\Request $request)
 	{
 		$qualifiedModuleName = $request->getModule(false);
-		$recordModel = Settings_WebserviceUsers_Record_Model::getInstanceById($request->getInteger('record', ''), $request->getByType('typeApi', 'Alnum'));
+		$typeApi = $request->getByType('typeApi', 'Alnum');
+		$recordModel = Settings_WebserviceUsers_Record_Model::getInstanceById($request->getInteger('record', ''), $typeApi);
 		$viewer = $this->getViewer($request);
 		$viewer->assign('TABLE_COLUMNS', static::$columnsToShow);
-		$viewer->assign('SESSION_HISTORY_ENTRIES', $recordModel->getUserSession());
+		$viewer->assign('SESSION_HISTORY_ENTRIES', $recordModel->getUserSession($typeApi));
```
Comment (mariuszkrzaczkowski) [inline modules/Settings/WebserviceUsers/views/ListViewSession.php]:
```suggestion
		$viewer->assign('SESSION_HISTORY_ENTRIES', $recordModel->getUserSession($container));
```

**Review by mariuszkrzaczkowski — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #14764 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/14764)

**Pull request merged**